apis/metastore/v1alpha1/service_types.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	refsv1beta1 "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	secretmanagerv1beta1 "github.com/GoogleCloudPlatform/k8s-config-connector/apis/secretmanager/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var MetastoreServiceGVK = GroupVersion.WithKind("MetastoreService")
+
+// +kcc:proto=google.cloud.metastore.v1.Secret
+type Secret struct {
+	// The relative resource name of a Secret Manager secret version, in the
+	//  following form:
+	//
+	//  `projects/{project_number}/secrets/{secret_id}/versions/{version_id}`.
+	// +kcc:proto:field=google.cloud.metastore.v1.Secret.cloud_secret
+	CloudSecret *secretmanagerv1beta1.SecretRef `json:"cloudSecret,omitempty"`
```

Comment:
I think the convention is to rename the field to SecretRef? Also we should update the field description to remove the format requirement because it is now a reference field.